src/flint/test/test.py

```diff
@@ -1815,6 +1814,204 @@ def test_fmpz_mod_dlog():
             assert g**x == a
 
 
+def _all_polys():
+    return [
+        # (poly_type, scalar_type, is_field)
+        (flint.fmpz_poly, flint.fmpz, False),
+        (flint.fmpq_poly, flint.fmpq, True),
+        (lambda *a: flint.nmod_poly(*a, 17), lambda x: flint.nmod(x, 17), True),
+    ]
+
+
+def test_polys():
+    for P, S, is_field in _all_polys():
+
+        assert P([S(1)]) == P([1]) == P(P([1])) == P(1)
+
+        assert raises(lambda: P([None]), TypeError)
```

Comment:
@GiacomoPope if we write the tests more like this then we don't need to duplicate the test code for different types and it also means that we can test that the different types have consistent method names and signatures.

Reply:
Yeah I agree with this. I think once #87 is merged I will take the time to address the tests before working on `fq` types.